core/src/test/java/com/datastax/dse/driver/internal/core/cql/reactive/TestSubscriber.java

```diff
@@ -81,7 +81,10 @@ public List<T> getElements() {
   }
 
   public void awaitTermination() {
-    Uninterruptibles.awaitUninterruptibly(latch, 1, TimeUnit.MINUTES);
+    boolean interrupted = Uninterruptibles.awaitUninterruptibly(latch, 1, TimeUnit.MINUTES);
```

Comment:
I believe this is incorrect. The returned value is the result of `latch.await()`, not a flag indicating that the thread was interrupted. The javadocs of `CountDownLatch.wait()` say:

> return true if the count reached zero and false if the waiting time elapsed before the count reached zero.

Also looking at the implementation of `awaitUninterruptibly` in 33.3, there is already a finally block where the thread is being interrupted:

```java
    } finally {
      if (interrupted) {
        Thread.currentThread().interrupt();
      }
    }
```

So I believe this code should be changed to:

```java
if (!Uninterruptibles.awaitUninterruptibly(latch, 1, TimeUnit.MINUTES)) {
  fail("subscriber not terminated");
}
```

Reply:
You are absolutely right. Not sure why I decided to change the code like this.